### PR TITLE
Updated EmojiMap; updated tooling to reflect new changes; API cleanup

### PR DESCRIPTION
**Pull request by diamondburned**
In this commit, the old "tools" directory was renamed to "cmd" to
reflect idiomatic Go package structures. The extractmap tool was also
updated to reflect new Discord changes.

Along with above updates, mapping.go was regenerated to reflect those
latest changes. It now includes skin tones as well as the latest Discord
emojis.

Example documentations are added into the tests, which should now add a
small "Example" popup in GoDoc.

A function was renamed to GetEntriesWithPrefix with slightly different
behaviors, all of which are documented. This IS a breaking change.

### CODE REVIEW

File: cmd/extractmap/main.go [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+// USE FILE:
+// https://discord.com/assets/b38205c8085075585265.js
+
+const goCode = `package discordemojimap
+
+var EmojiMap = map[string]string {
+%s}
+`
+
+var emojiJSONRegex = regexp.MustCompile(`'{"people":.*}'`)
```
Comment (Bios-Marcel) [inline cmd/extractmap/main.go]: Could you please explain this? I can't quite infer the meaning ^^

Comment (diamondburned) [reply cmd/extractmap/main.go]: The new JS file has some hard-coded JSON that it parses on runtime by Webpack.

Comment (Bios-Marcel) [reply cmd/extractmap/main.go]: And something with "people" happens to be at the beginning of that map? If so, please add that as a comment

**Comment by Bios-Marcel** [comment]
:+1:

**Comment by Bios-Marcel** [comment]
I pushed a v2

**Pull request merged**